game/src/story/write_text.asm

```diff
@@ -86,22 +87,45 @@ PutCharLoop:: ;1d11, things jump to here after the control code
   rst $8
   add hl, bc
   call GetNextChar
-  ld a, [hl]
-  cp $4f
-  jp z, Char4F
-  cp $4e
-  jp z, Char4E
-  cp $4d
-  jp z, Char4D
-  cp $4c
-  jp z, Char4C
-  cp $4b
-  jp z, Char4B
-  cp $4a
-  jp z, Char4A
-  jp SkipSpace
+
+  ; Store the current character index as well as potential arguments for control codes in WRAM. Point hl to WRAM location.
+
+  ld b, h
+  ld c, l
+  ld hl, VWFCurrentLetter
+  ld a, [bc]
+  ld [hli], a
+  inc bc
+  ld a, [bc]
+  ld [hli], a
+  inc bc
+  ld a, [bc]
+  ld [hld], a
+  dec l
+
+  ; Switch to the bank where the vwf font is located.
+
+  ld a, BANK(VWFFont)
```

Comment:
You can take advantage of the framework in patch/hack.asm to do this. 

As long as your routine is in bank $24, you can add it to the HackPredefTable and reference it's index ($c is the next open one):

```
ld a, $c
rst $8
```
which will handle bank swapping and returning.

Reply:
Never mind, after looking more closely, this is necessary because you loop this to pull information.

That's fine, no change needed.